circuits/transfer/src/withdraw_from_obfuscated.rs

```diff
@@ -71,41 +146,120 @@ impl Circuit for WithdrawFromObfuscatedCircuit {
         &mut self,
         composer: &mut StandardComposer,
     ) -> Result<(), PlonkError> {
-        let mv = composer.add_input(self.message_value.into());
-        let mb = composer.add_input(self.message_blinder.into());
+        // 1. commitment(Ic, Iv, Ib, 64)
+        let iv = composer.add_input(self.input_value.into());
+        let ib = composer.add_input(self.input_blinder.into());
+        let input_commitment = gadgets::commitment(composer, iv, ib);
+
+        composer.range_gate(iv, 64);
+        composer.assert_equal_public_point(
+            input_commitment,
+            self.input_commitment.into(),
+        );
+
+        // 2. commitment(Cc, Cv, Cb, 64)
+        let cv = composer.add_input(self.change_value.into());
+        let cb = composer.add_input(self.change_blinder.into());
+        let change_commitment = gadgets::commitment(composer, cv, cb);
 
+        composer.range_gate(cv, 64);
+        composer.assert_equal_public_point(
+            change_commitment,
+            self.change_commitment.into(),
+        );
+
+        // 3. commitment(Oc, Ov, Ob, 64)
         let ov = composer.add_input(self.output_value.into());
         let ob = composer.add_input(self.output_blinder.into());
+        let output_commitment = gadgets::commitment(composer, ov, ob);
 
-        // 1. commitment(Mc,mv,mb)
-        let commitment = gadgets::commitment(composer, mv, mb);
+        composer.range_gate(ov, 64);
         composer.assert_equal_public_point(
-            commitment,
-            self.message_commitment.into(),
+            output_commitment,
+            self.output_commitment.into(),
+        );
+
+        // 4. (pa, pb) := selectPair(Cx, I, Cp, Cs)
+        let identity = Point::identity(composer);
```

Comment:
@CPerezz would it be an idea to add `Point::identity()` to the composer API? Just curious, because it seems different in nature to the rest of the methods.

Reply:
Will be fixed in https://github.com/dusk-network/plonk/pull/575